java8/src/main/java/com/dslplatform/json/processor/ConverterTemplate.java

```diff
@@ -425,7 +441,33 @@ void fromObject(final StructInfo si, final String className) throws IOException
 		code.append("\t}\n");
 	}
 
-	private void writeObject(final String className, List<AttributeInfo> sortedAttributes) throws IOException {
+	private void writeDiscriminator(final StructInfo si) throws IOException {

```

Comment:
I thought a bit about what you are trying to do here and I think I understand.
It makes sense to always serialize discriminator when it's defined on an object.
The current deserializeDiscriminator was kind of useless on non abstract things anyway, and this would make it useful.
In that regard it would be better if I called it just discriminator (it's still new so I'm fine with making that breaking change).
But in that case, discriminator should be written outside of writeContentFull, since that is reused by the mixin thing.

Reply:
Yes, might be the MixinFormatter is better place where we can write deserializer.

Reply:
MixinFormatter already does the expected thing. If we want to have this new feature to always put discriminator even on classes, you should do it outside of writeContent method, eg, here: https://github.com/ngs-doo/dsl-json/blob/master/java8/src/main/java/com/dslplatform/json/processor/ConverterTemplate.java#L442 and https://github.com/ngs-doo/dsl-json/blob/master/java8/src/main/java/com/dslplatform/json/processor/ConverterTemplate.java#L599